---
- name: Create a snapshot, erroring if one already exists at path
  block:
  - name: Create a snapshot named 'snapshot_error'
    community.general.btrfs_subvolume:
      automount: Yes
      name: "/snapshot_error"
      snapshot_source: "/"
      snapshot_conflict: "error"
      device: "{{ btrfs_subvolume_target_device | default(omit) }}"
      state: "present"
    register: result
  - name: Snapshot 'snapshot_error' created
    ansible.builtin.assert:
      that:
      - result is changed

  - name: Create a snapshot named 'snapshot_error' (no idempotency)
    community.general.btrfs_subvolume:
      automount: Yes
      name: "/snapshot_error"
      snapshot_source: "/"
      snapshot_conflict: "error"
      device: "{{ btrfs_subvolume_target_device | default(omit) }}"
      state: "present"
    register: result
    ignore_errors: true
  - name: Snapshot 'snapshot_error' created (no idempotency)
    ansible.builtin.assert:
      that:
      - result is not changed

- name: Cleanup created snapshot
  community.general.btrfs_subvolume:
    automount: Yes
    name: "/snapshot_error"
    device: "{{ btrfs_subvolume_target_device | default(omit) }}"
    state: "absent"
